homeassistant/components/netgear_lte/sensor.py

```diff
@@ -1,41 +1,49 @@
 """Support for Netgear LTE sensors."""
+import logging
+
 import attr
-import voluptuous as vol
 
-from homeassistant.components.sensor import PLATFORM_SCHEMA
-from homeassistant.const import CONF_HOST, CONF_SENSORS
+from homeassistant.components.sensor import DOMAIN
 from homeassistant.exceptions import PlatformNotReady
-import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity import Entity
 
-from ..netgear_lte import DATA_KEY
+from ..netgear_lte import CONF_MONITORED_CONDITIONS, DATA_KEY
 
 DEPENDENCIES = ['netgear_lte']
 
+_LOGGER = logging.getLogger(__name__)
+
 SENSOR_SMS = 'sms'
 SENSOR_USAGE = 'usage'
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_HOST): cv.string,
-    vol.Required(CONF_SENSORS): vol.All(
-        cv.ensure_list, [vol.In([SENSOR_SMS, SENSOR_USAGE])]),
-})
+DEFAULT_SENSORS = [SENSOR_SMS, SENSOR_USAGE]
 
 
 async def async_setup_platform(
         hass, config, async_add_entities, discovery_info):
     """Set up Netgear LTE sensor devices."""
-    modem_data = hass.data[DATA_KEY].get_modem_data(config)
+    if discovery_info is None:
+        return
+
+    modem_data = hass.data[DATA_KEY].get_modem_data(discovery_info)
 
     if not modem_data:
         raise PlatformNotReady
 
+    sensor_conf = discovery_info[DOMAIN]
+    if sensor_conf:
+        monitored_conditions = sensor_conf[CONF_MONITORED_CONDITIONS]
+    else:
+        monitored_conditions = DEFAULT_SENSORS
```

Comment:
We can move the default to the voluptuous schema, right?

Reply:
I would like a suggestion on how to structure that since it seems to me that the list of sensor names belongs in the sensor module which should not be imported into the component.

Reply:
They could be defined in a const submodule and imported from there to the package module and to the sensor module.

Reply:
Thanks 👍. You know, I prepared this PR in the old times and rejected that setup as being too ugly. Somehow that idea stuck with me after the Great Migration but now it doesn't look so bad :)